src/3d/mesh/qgsmesh3dentity_p.cpp

```diff
@@ -53,14 +53,15 @@ void QgsMesh3dEntity::buildGeometry()
   }
 
   mesh->setGeometry( new QgsMesh3dGeometry( mTriangularMesh,
+                     mMapSettings.origin(),
                      mExtent,
                      mSymbol.verticaleScale(),
                      mesh ) );
 
   addComponent( mesh );
 
   Qt3DCore::QTransform *tform = new Qt3DCore::QTransform;
-  tform->setTranslation( QVector3D( float( -mMapSettings.origin().x() ), 0, float( mMapSettings.origin().y() ) ) );
+  tform->setTranslation( QVector3D( 0, 0, 0 ) ) ;
```

Comment:
If we don't apply any translation through the transform anymore, we could get rid of the whole qtransform component... (also in the code below)